degirum_tools/video_support.py

```diff
@@ -104,12 +111,17 @@ def get_props(stream: cv2.VideoCapture) -> tuple:
             return get_props(stream)
 
 
-def video_source(stream: cv2.VideoCapture) -> Generator[np.ndarray, None, None]:
+def video_source(
+    stream: cv2.VideoCapture, fps: Optional[float] = None, frame_decimate: Optional[bool] = False
+) -> Generator[np.ndarray, None, None]:
     """Generator function, which returns video frames captured from given video stream.
     Useful to pass to model batch_predict().
 
     stream - video stream context manager object returned by open_video_stream()
-
+    fps - optional fps cap. If greater than the actual FPS, it will do nothing.
+       If less than the current fps, it will decimate frames accordingly.
+    frame_decimate - If fps is set, and frame decimation is True, frames are decimated
```

Comment:
It will be much cleaner API, if you will remove this decimate parameter from API at all and will deduce it inside `video_source()` function.

According to chatGPT:

cap.get(cv2.CAP_PROP_FRAME_COUNT) retrieves the number of frames in the video file. If it's a live camera feed, this value would be 0.

So, in order to deduce, which FPS control mode to use (decimation or throttling) you may analyze `cv2.CAP_PROP_FRAME_COUNT` inside `video_source()`: if it is zero, then you have live/realtime stream and you have to use throttling. Otherwise, use decimation.